projects/step-frontend/src/lib/modules/timeseries/modules/legacy/components/time-series-dashboard/time-series-dashboard.component.ts

```diff
@@ -96,22 +97,17 @@ export class TimeSeriesDashboardComponent implements OnInit, OnDestroy {
     activeFilters: FilterBarItem[],
   ) {
     const contextualFilters = Object.keys(contextualParams).map((key) => {
-      const fieldType = this.getFilterFieldType(key);
-      const isEntityFilter = [FilterBarItemType.PLAN, FilterBarItemType.TASK, FilterBarItemType.EXECUTION].includes(
-        fieldType,
-      );
-      const searchValue = contextualParams[key];
-      return {
-        isHidden: !this.settings.showContextualFilters,
-        removable: true,
-        label: this.getFilterFieldLabel(key),
-        type: fieldType,
-        attributeName: key,
-        freeTextValues: fieldType === 'FREE_TEXT' ? [searchValue] : [],
-        searchEntities: isEntityFilter ? [{ searchValue: searchValue }] : [],
-        isLocked: false,
-        exactMatch: true,
-      } as FilterBarItem;
+      const filterBarItem = FilterUtils.createEmptyFilterTextItem(key);
+
+      const searchValue: string = contextualParams[key]!;
+
+      filterBarItem.isHidden = !this.settings.showContextualFilters;
+      filterBarItem.removable = true;
+      filterBarItem.freeTextValues = filterBarItem.type === 'FREE_TEXT' ? [searchValue] : [];
```

Comment:
```suggestion
      filterBarItem.freeTextValues = (filterBarItem.type === 'FREE_TEXT' ? [searchValue] : []);
```

Reply:
@neogucky why do you need this?

Reply:
This is not needed but improves readability, I really like brackets around ?-operators